functions/server_version.pp

```diff
@@ -0,0 +1,24 @@
+# Authoritatively determine the puppet server version and return `undef` if one
+# could not be determined.
+#
+# @return [String]
+#   The puppet server version
+#
+function pupmod::server_version {
```

Comment:
use pick?

Reply:
Sort of. Have to have the `defined` in there, but otherwise, sure.